otelcolbuilder/cmd/collector_windows.go

```diff
@@ -246,19 +247,17 @@ func updateSettingsUsingFlags(set *otelcol.CollectorSettings, flags *flag.FlagSe
 }
 
 func newDefaultConfigProviderSettings(uris []string) otelcol.ConfigProviderSettings {
-	converterSet := confmap.ConverterSettings{}
-	providerSet := confmap.ProviderSettings{}
 	return otelcol.ConfigProviderSettings{
 		ResolverSettings: confmap.ResolverSettings{
 			URIs: uris,
-			Providers: makeMapProvidersMap(
-				fileprovider.NewWithSettings(providerSet),
-				envprovider.NewWithSettings(providerSet),
-				yamlprovider.NewWithSettings(providerSet),
-				httpprovider.NewWithSettings(providerSet),
-				httpsprovider.NewWithSettings(providerSet),
-			),
-			Converters: []confmap.Converter{expandconverter.New(converterSet)},
+			ProviderFactories: []confmap.ProviderFactory{
+				fileprovider.NewFactory(),
+				envprovider.NewFactory(),
+				yamlprovider.NewFactory(),
+				httpprovider.NewFactory(),
+				httpsprovider.NewFactory(),
```

Comment:
Could you compare with linux collector? I see we are missing `globprovider` here

Reply:
I'm not sure if we need to have `collector_windows.go` in our repo, main_windows.go has this form:
```
// Code generated by "go.opentelemetry.io/collector/cmd/builder". DO NOT EDIT.

//go:build windows

package main

import (
	"errors"
	"fmt"
	"golang.org/x/sys/windows"
	"golang.org/x/sys/windows/svc"
	"go.opentelemetry.io/collector/otelcol"
)

func run(params otelcol.CollectorSettings) error {
	// No need to supply service name when startup is invoked through
	// the Service Control Manager directly.
	err := UseCustomConfigProvider(&params)
	if err != nil {
		return err
	}
	cmd := otelcol.NewCommand(params)
	// this is only here so that the flag can be recognized by the upstream parser and displayed in help
	cmd.Flags().StringVarP(&opAmpConfig, "remote-config", "", "", "path to opamp config file")
	if err := svc.Run("", NewSvcHandler(params)); err != nil {
		if errors.Is(err, windows.ERROR_FAILED_SERVICE_CONTROLLER_CONNECT) {
			// Per https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicectrldispatchera#return-value
			// this means that the process is not running as a service, so run interactively.
			return runInteractive(params)
		}

		return fmt.Errorf("failed to start collector server: %w", err)
	}

	return nil
}

```

[UseCustomConfigProvider](https://github.com/SumoLogic/sumologic-otel-collector/blob/a840af0fff60965734c246dc1c198ecc2cbed754/otelcolbuilder/cmd/configprovider.go#L42) uses this [NewConfigProviderSettings](https://github.com/SumoLogic/sumologic-otel-collector/blob/a840af0fff60965734c246dc1c198ecc2cbed754/otelcolbuilder/cmd/configprovider.go#L97) which has globprovider 🤷 

We have test for globprovider - [here](https://github.com/SumoLogic/sumologic-otel-collector/blob/a840af0fff60965734c246dc1c198ecc2cbed754/otelcolbuilder/cmd/collector_config_test.go#L82) and tests are executed on windows

Reply:
issue for checking if we need to have `collector_windows.go` https://github.com/SumoLogic/sumologic-otel-collector/issues/1565

Reply:
I'm not sure if this code is used right now.

Reply:
ok, I'll add `globprovider` here